driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -671,22 +753,36 @@ private boolean addConnectionIfUnderMaximum(int shardId, ConnectionTasksSharedSt
       if (newConnection == null) {
         if (!host.convictionPolicy.canReconnectNow()) {
           open[shardId].decrementAndGet();
-          return false;
+          return ConnectionResult.SHOULD_RETRY;
```

Comment:
Why not make it `ConnectionResult.FAILED`, preserving the old behavior?

Reply:
New behavior is just better.